python/ray/_private/worker.py

```diff
@@ -2027,6 +2027,29 @@ def connect(
         runtime_env.pop("excludes", None)
         job_config.set_runtime_env(runtime_env)
 
+    if mode == SCRIPT_MODE:
```

Comment:
Can you help me understand why this block needed to move before core worker initializatilon?

Reply:
there are two (relevant) types of workers: drivers and task/actor execution workers.
the job_config flow from drivers -> raylet (worker_pool) -> executor-workers, both implicitly happens through coreworker [constructor](https://sourcegraph.com/github.com/ray-project/ray/-/blob/src/ray/core_worker/core_worker.cc?L164) `worker.core_worker = ray._raylet.CoreWorker`, where driver set the job_config to raylet and executor receive the job_config from the raylet.

That's why we need to change it before core_worker initialization, as otherwise the job_config has already being sent to raylet.

Reply:
btw how executor-worker get job_config will change by https://github.com/ray-project/ray/pull/31375 soon